cmd/snapd-generator/main.c

```diff
@@ -30,30 +31,168 @@
 #include "../libsnap-confine-private/mountinfo.h"
 #include "../libsnap-confine-private/string-utils.h"
 
-static sc_mountinfo_entry *find_root_mountinfo(sc_mountinfo * mounts)
+static sc_mountinfo_entry *find_dir_mountinfo(const char *mnt_dir)
 {
+	// Load /proc/1/mountinfo so that we can inspect the root filesystem.
+	sc_mountinfo *mounts SC_CLEANUP(sc_cleanup_mountinfo) = NULL;
```

Comment:
This looks wrong. This will iterate over the list inside sc_mountinfo and free each item, finally freeing the structure itself. Yet we want to return an entry from the list.

Reply:
Fixed now